### PR TITLE
JDK-8291823: IGV: Fix "Save selected groups"

### PR DESCRIPTION
**Pull request by tobiasholenstein**
In IGV the button `Save selected groups to XML file...` is broken and never activated. 

*Before this fix* 
Bellow you can see the Outline with two selected groups, but the `Save selected groups to XML file...` button does not work:
before_groups_selected

*After this fix* 
Now the `Save selected groups to XML file...` button works if at least one groups is selected:
groups_selected

*After this fix* 
The `Save selected groups to XML file...` button is greyed out if no groups are selected:
no_groups_selected

*Before this fix* 
The `Save all groups to XML` and the `Remove all graphs and groups` buttons where available even if no graphs or groups were loaded:
before_no_groups

*After this fix* 
The `Save all groups to XML` and the `Remove all graphs and groups` buttons are greyed out if no graphs or groups are loaded:
no_groups

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8291823](https://bugs.openjdk.org/browse/JDK-8291823): IGV: Fix "Save selected groups"

### Reviewers
 * [Roberto Castañeda Lozano](https://openjdk.org/census#rcastanedalo) (@robcasloz - **Reviewer**)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9730/head:pull/9730` \
`$ git checkout pull/9730`

Update a local copy of the PR: \
`$ git checkout pull/9730` \
`$ git pull https://git.openjdk.org/jdk pull/9730/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9730`

View PR using the GUI difftool: \
`$ git pr show -t 9730`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9730.diff">https://git.openjdk.org/jdk/pull/9730.diff</a>

</details>

### CODE REVIEW

File: src/utils/IdealGraphVisualizer/Coordinator/src/main/java/com/sun/hotspot/igv/coordinator/actions/ImportAction.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2008, 2016, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2008, 2022, Oracle and/or its affiliates. All rights reserved.
```
Comment (robcasloz) [inline src/utils/IdealGraphVisualizer/Coordinator/src/main/java/com/sun/hotspot/igv/coordinator/actions/ImportAction.java:2]: Please do not update the copyright line if there are no other changes in the file.

**Review by robcasloz — CHANGES_REQUESTED** [review]
Thanks for working on this @tobiasholenstein!

I tested the changeset manually and found that, after selecting all groups listed in the Outline window and clicking on `Remove selected graphs and groups`, we end up with an empty outline where the `Save selected groups to XML file...` and `Remove selected graphs and groups` buttons are still enabled:

empty-outline

Nit: to ease reviewing (and backporting, etc.), try to minimize the number of empty lines added/removed that are not directly related to the changeset.

File: src/utils/IdealGraphVisualizer/Coordinator/src/main/java/com/sun/hotspot/igv/coordinator/actions/ImportAction.java (hunk 1 above)
Comment (tobiasholenstein) [reply src/utils/IdealGraphVisualizer/Coordinator/src/main/java/com/sun/hotspot/igv/coordinator/actions/ImportAction.java]: Thanks, I reverted this

**Comment by tobiasholenstein** [comment]
> Thanks for working on this @tobiasholenstein!
> 
> I tested the changeset manually and found that, after selecting all groups listed in the Outline window and clicking on `Remove selected graphs and groups`, we end up with an empty outline where the `Save selected groups to XML file...` and `Remove selected graphs and groups` buttons are still enabled:
> 
> empty-outline
> 
> Nit: to ease reviewing (and backporting, etc.), try to minimize the number of empty lines added/removed that are not directly related to the changeset.

Good catch! The problem here was that all the `FolderNodes` are in an invisible root `FolderNode` that gets selected when all its children are removed. I fixed it by checking that the selected `FolderNode` is not the root.

**Review by robcasloz — APPROVED** [review]
Thanks for fixing the enable/disable inconsistency, looks good to me now.

**Review by vnkozlov — APPROVED** [review]
Good.

**Review by TobiHartmann — APPROVED** [review]
Looks good.

**Comment by tobiasholenstein** [comment]
Thanks @robcasloz , @vnkozlov and @TobiHartmann for the reviews!

**Pull request closed without merging**